### PR TITLE
debug: Add VTune agent

### PR DESCRIPTION
**Pull request by temerkhanov**
Add VTune agent to the debug container

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>

### CODE REVIEW

File: pkg/debug/Dockerfile.in [hunk 1]
```diff
@@ -37,6 +38,41 @@ RUN for patch in fix-musl-sc_long_bit.patch wrapper-for-basename.patch 155652295
  make -C src RPM_OPT_FLAGS=-DNONLS static &&\
  cp src/lshw-static /lshw && strip /lshw
 
+#FROM intel/oneapi-basekit as vtune-source
```
Comment (rvs) [inline pkg/debug/Dockerfile.in:41]: do we need to keep this comment?

**Review by rvs — CHANGES_REQUESTED** [review]
This looks mostly fine (aside from one small nit in the Dockerfile) my question, however, is that whether vtune agent is useful outside of... well... very developer focused tuning.

IOW, are there any scenarios that  a support person can use it (lets say in the way support person can use D-Trace on a production system). If not -- I suggest we create a separate container for it called eve-tune.

File: pkg/debug/Dockerfile.in (hunk 1 above)
Comment (temerkhanov) [reply pkg/debug/Dockerfile.in:41]: We might as well skip it - the binary version depends on a full glibc

Comment (rvs) [reply pkg/debug/Dockerfile.in:41]: Yeah -- in that case -- let make a separate container that is specifically there for VTune purposes -- lets not pollute debug one @temerkhanov

**Review by rvs — CHANGES_REQUESTED** [review]
With this latest revelation around VTune requiring full glibc -- lets just move to a dedicated container here.

File: pkg/debug/Dockerfile.in (hunk 1 above)
Comment (temerkhanov) [reply pkg/debug/Dockerfile.in]: That is the binary agent version which depends on glibc (and pulls a really huge container during build). The one added by this patch is built from source and is OK with musl

**Comment by temerkhanov** [comment]
This version builds the agent from source and is fine with musl

**Comment by rvs** [comment]
> This version builds the agent from source and is fine with musl

Sure, but it seems to make the container quite a bit bigger and you still didn't answer my question on whether we would need it in any other scenario @temerkhanov

**Comment by temerkhanov** [comment]
Moved to a separate container

File: images/rootfs.yml.in [hunk 2]
```diff
@@ -66,6 +66,9 @@ services:
      image: XENTOOLS_TAG
      cgroupsPath: /eve/services/xen-tools
      oomScoreAdj: -999
+   - name: vtune
```
Comment (rvs) [inline images/rootfs.yml.in:69]: Let's drop it from the default builds for now and move it to a special profile (see how we're doing `rootfs-acrn.yml.in.patch` for example) -- the rest looks good @temerkhanov

**Review by rvs — CHANGES_REQUESTED** [review]

File: pkg/vtune/Dockerfile.in [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#hadolint ignore=DL3006
+FROM DEBUG_TAG as musl-build
+#hadolint ignore=DL3006
+FROM NEW_KERNEL_TAG as kernel-build
+FROM alpine:3.12 as vtune-build
+
+RUN apk add --no-cache curl=7.69.1-r3 tar=1.32-r1 make=4.3-r0 linux-headers=5.4.5-r1 patch=2.7.6-r6 g++=9.3.0-r2 \
+                       coreutils=8.32-r0 git=2.26.2-r0
```
Comment (dautovri) [inline pkg/vtune/Dockerfile.in:8]: Needs to change git version to git=2.26.3-r0

**Comment by rouming** [comment]
I don't think we are going to continue work on that. @eriknordmark could you please close this with the "stalled" label?

**Comment by eriknordmark** [comment]
Closing as stalled

**Pull request closed without merging**